### ISSUE TITLE
difference in import-certs jenkins and SQ (update-ca-trust vs update-ca-certificates)

### ISSUE THREAD

**Issue by clemensutschig** [opendevstack/ods-core#1071]
**Describe the bug**
[(jenkins master · opendevstack/ods-core (github.com)](https://github.com/opendevstack/ods-core/blob/master/jenkins/master/import_certs.sh#L21)

uses `update-ca-trust` (which works) resulting in
````
[2021-11-19T11:29:21.502Z] [0;32m    amazon-ebs: Setting up certificates from APP_DNS=docker-registry-default.ocp.odsbox.lan ...[0m
[2021-11-19T11:29:21.502Z] [0;32m    amazon-ebs: Importing DNS=docker-registry-default.ocp.odsbox.lan PORT=443[0m
[2021-11-19T11:29:21.502Z] [0;32m    amazon-ebs: [91mdepth=1 CN = openshift-signer@1637320235[0m
[2021-11-19T11:29:21.502Z] [0;32m    amazon-ebs: verify error:num=19:self signed certificate in certificate chain[0m
[2021-11-19T11:29:21.502Z] [0;32m    amazon-ebs: [0m[91mDONE[0m
[2021-11-19T11:29:22.067Z] [0;32m    amazon-ebs: [0m[91mWarning: use -cacerts option to access cacerts keystore[0m
[2021-11-19T11:29:22.633Z] [0;32m    amazon-ebs: [0m[91mCertificate was added to keystore[0m
[2021-11-19T11:29:23.569Z] [0;32m    amazon-ebs: [0mDone with certificate setup[0m
[2021-11-19T11:29:24.139Z] [0;32m    amazon-ebs: Removing intermediate container 28ace807bc02[0m
``

while

[(SQ master · opendevstack/ods-core (github.com)](https://github.com/opendevstack/ods-core/blob/master/sonarqube/docker/import_certs.sh#L21) 

uses `update-ca-certificates` (which fails with a core dump, some cases) resulting in

```
[2021-11-19T11:26:07.697Z] [0;32m    amazon-ebs: Step 11/37 : RUN import_certs.sh[0m
[2021-11-19T11:26:07.697Z] [0;32m    amazon-ebs:  ---> Running in 3129087e5fef[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: Setting up certificates from APP_DNS=docker-registry-default.ocp.odsbox.lan ...[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: Importing DNS=docker-registry-default.ocp.odsbox.lan PORT=443[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: [91mdepth=1 CN = openshift-signer@1637320235[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: verify error:num=19:self signed certificate in certificate chain[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: verify return:1[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: depth=1 CN = openshift-signer@163[0m[91m7320235[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: verify return:1[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: [0m[91mdepth=0 CN = *.ocp.odsbox.lan[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: verify return:1[0m
[2021-11-19T11:26:07.955Z] [0;32m    amazon-ebs: [0m[91mDONE[0m
[2021-11-19T11:26:08.214Z] [0;32m    amazon-ebs: [0m[91mWarning: use -cacerts option to access cacerts keystore[0m
[2021-11-19T11:26:08.780Z] [0;32m    amazon-ebs: [0m[91mCertificate was added to keystore[0m
[2021-11-19T11:26:08.780Z] [0;32m    amazon-ebs: [0m[91mWARNING: ca-certificates.crt does not contain exactly one certificate or CRL: skipping[0m
[2021-11-19T11:26:08.780Z] [0;32m    amazon-ebs: [0m[91mWARNING: ca-cert-docker-registry-default.ocp.odsbox.lan.pem does not contain exactly one certificate or CRL: skipping[0m
[2021-11-19T11:26:08.780Z] [0;32m    amazon-ebs: [0mDone with certificate setup[0m
```

**Comment by michaelsauter** [comment]
I realised the test does not cover this as it passed `--build-arg idpDns=""` to `docker build`, resulting in the code in question to be skipped. 

The SonarQube image is based on `adoptopenjdk/openjdk11:x86_64-alpine-jre-11.0.10_9`, which seems to be a pretty specific tag. I wonder what changed .... I am going to rebuild the image locally to verify.

**Comment by michaelsauter** [comment]
After talking this over with @clemensutschig, we reached the following conclusions:

* SQ uses `update-ca-certificates` as it is Alpine based. Jenkins uses `update-ca-trust` as it is RHEL7/UBI8 based
* `update-ca-certificates` works well in the devenv / ODS AMI CI build (with a self-signed certificate)

Therefore the encountered error must result from the combination of the specific certificate in use and `update-ca-certificates`. For some reason, that certificate does not cause `update-ca-trust` to fail.

A side remark on a theory we pursued for a bit: the base image (`adoptopenjdk/openjdk11:x86_64-alpine-jre-11.0.10_9`) is already 7 months old: https://hub.docker.com/layers/adoptopenjdk/openjdk11/x86_64-alpine-jre-11.0.10_9/images/sha256-a97add2e5cc7588fb7bf67a9be4815dc0acabe914af123fbac4e028a0c3140bf?context=explore. However, the `Dockerfile` runs `apk update && apk add --no-cache ca-certificates` so the package should be up-to-date, therefore this shouldn't be the issue ...
